src/Symfony/Bundle/TwigBundle/CacheWarmer/TemplateCacheWarmer.php

```diff
@@ -26,21 +26,26 @@
  */
 class TemplateCacheWarmer implements CacheWarmerInterface, ServiceSubscriberInterface
 {
-    private ContainerInterface $container;
     private Environment $twig;
-    private iterable $iterator;
 
-    public function __construct(ContainerInterface $container, iterable $iterator)
+    public function __construct(private ContainerInterface $container, private iterable $iterator, private string $cacheFolder = 'twig')
     {
-        // As this cache warmer is optional, dependencies should be lazy-loaded, that's why a container should be injected.
-        $this->container = $container;
-        $this->iterator = $iterator;
+        if (\func_num_args() < 3) {
+            trigger_deprecation('symfony/twig-bundle', '7.1', 'The "string $cacheFolder" argument of "%s()" method will not be optional anymore in version 8.0, not defining it is deprecated.', __METHOD__);
+        }
     }
 
     public function warmUp(string $cacheDir, ?string $buildDir = null): array
     {
+        if (!$buildDir) {
+            return [];
+        }
+
         $this->twig ??= $this->container->get('twig');
 
+        $originalCache = $this->twig->getCache();
+        $this->twig->setCache($buildDir.\DIRECTORY_SEPARATOR.$this->cacheFolder);
```

Comment:
This assumes the cache instance is configured to store files in this path, but the Twig environment could be configured differently.

To avoid having too much knowledge in this dumb warmup class, what do you think of adding an interface `WarmableLoader` with a method `warmup`. It would called only here on loaders that implement this interface.

Reply:
I think the same, the way this is done is prone to misconfiguration. But, it's not the only CacheWarmer to be constructed on that principle, and it would deserve a proper, global solution IMHO.

That being said, the Twig environment is configured by the same Bundle, so even if the configuration is not really enforced, the Bundle is still in control of the two sides.